examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -243,4 +250,21 @@ public Address receivingAddress() {
     static String getPrefix(BitcoinNetwork network) {
         return String.format("forwarding-service-%s", network.toString());
     }
+
+    static class ForwardingCoinSelector implements CoinSelector {
+        private final Sha256Hash forwardTxId;
+
+        public ForwardingCoinSelector(Transaction transactionToForward) {
+            this.forwardTxId = transactionToForward.getTxId();
+        }
+
+        @Override
+        public CoinSelection select(Coin target, List<TransactionOutput> candidates) {
+            var selected =
+                    candidates.stream()
+                            .filter(output -> output.getParentTransactionHash().equals(forwardTxId))
+                            .collect(Collectors.toSet());
+            return new CoinSelection(selected);
+        }
+    }
```

Comment:
This can be simplified to a lambda after we complete PR #2431.

Reply:
This isn't a suggested change, just a comment. The GitHub review UI is a little bit awkward.